xrpl4j-core/src/main/java/org/xrpl/xrpl4j/crypto/keys/bc/BcKeyUtils.java

```diff
@@ -100,8 +100,10 @@ public static PrivateKey toPrivateKey(final Ed25519PrivateKeyParameters ed25519P
   public static PrivateKey toPrivateKey(final ECPrivateKeyParameters ecPrivateKeyParameters) {
     // Convert the HEX representation of the BigInteger into bytes.
     byte[] privateKeyBytes = BaseEncoding.base16().decode(ecPrivateKeyParameters.getD().toString(16).toUpperCase());
-    final UnsignedByteArray privateKeyUba = UnsignedByteArray.of(privateKeyBytes);
-    return PrivateKey.of(privateKeyUba);
+    return PrivateKey.of(
+      UnsignedByteArray.of(PrivateKey.SECP256K1_PREFIX) // <-- Per #486, always append 0x00 to secp256k1 private keys
+        .append(UnsignedByteArray.of(privateKeyBytes))
```

Comment:
Are we sure `ecPrivateKeyParameters.getD()` always returns 32 bytes? Is it worth just double checking here so we don't accidentally add a 0x00 prefix to an already prefixed array?

Reply:
From some cursory tests, it seems to always yield 32 bytes. But just in case I added pad-check code.